resolver/dns/dns_resolver.go

```diff
@@ -24,10 +24,27 @@
 package dns
 
 import (
+	"time"
+
 	"google.golang.org/grpc/internal/resolver/dns"
 	"google.golang.org/grpc/resolver"
 )
 
+// SetResolvingTimeout sets the maximum duration for DNS resolution requests.
+//
+// This function affects the global timeout used by all channels using the DNS name resolver scheme.
+//
+// It must be called only at application startup, before any gRPC calls are made.
+// Modifying this value after initialization is not thread-safe.
+//
+// The default value is 30 seconds. Setting the timeout too low may result
+// in premature timeouts during resolution, while setting it too high may
+// lead to unnecessary delays in service discovery. Choose a value appropriate
+// for your specific needs and network environment.
```

Comment:
Similar nit formatting suggestion

```suggestion
// SetResolvingTimeout sets the maximum duration for DNS resolution requests.
//
// This function affects the global timeout used by all channels using the DNS
// name resolver scheme.
//
// It must be called only at application startup, before any gRPC calls are
// made. Modifying this value after initialization is not thread-safe.
//
// The default value is 30 seconds. Setting the timeout too low may result in
// premature timeouts during resolution, while setting it too high may lead to
// unnecessary delays in service discovery. Choose a value appropriate for your
// specific needs and network environment.
```